### DISCUSSION TITLE
build gear template switch Module

### DISCUSSION THREAD

**Discussion by Taschenbuch** [blish-hud/Blish-HUD#881] [Module Ideas]
# gw2 ingame situation
guild wars 2 has equipment tabs and gear tabs. They can be switched ingame:
- in the hero panel
- with hotkeys (by default no hotkeys are configured. Each tab requires one hotkey. e.g. equip tab 2 and build tab 2 both require their own hotkeys. Thus you need at least 2 key presses to switch build and gear template. e.g. my old setup was like that F1 to F3 change equip template tabs, F1+alt to F3+alt change build template tabs)
- by clicking on the bottom left icon (right or left click?) to open a list of avaiable build and equipment templates for quick access
![image](https://user-images.githubusercontent.com/43114787/233177979-04e46b60-d073-4681-b214-4c8f3c2ae48d.png)
- details wiki: https://wiki.guildwars2.com/wiki/Template

# Possible module features
- in the module settings tell the user to add keybindings for the build tabs and gear tabs in gw2 settings AND in the same hotkeys in the module settings.
- the modules adds a way to switch gear AND build template at the same time. possible options
  - the option to show a cornerIcon/menu icon. By left clicking on that a contextmenu pops up (dont use right click! see blish ui/ux guide for details). The contextmenu shows available builds like "heal firebrand" "wvw heal scrapper" "rifle scrapper" for the currently played character. By clicking on one of the builds the build/gear tabs for that build will be selected 
  - the option to setup custom hotkeys for builds. pressing the hotkey for a build will switch gear AND build template at once. This may require visual feedback to the user which build has been chosen. maybe by showing a blish hud alert or whatever it was called.
  - the option to add buttons in the ui like a "heal firebrand" button. you can click such a button to switch to the appropriate build (gear + build). the button position and maybe size etc. should be customizable
  - the option to use a radial menu like mount module and character module to switch builds
- show currently selected buildname permanently in the UI (not sure if that can be detected ingame. Might be worth checking the current template module for that)

# Annotations
  - builds are character specific. So every character requires its individual setup (not just class/spec) .
  - the build name and which build tab and which gear tab belong to the build may need to be fully customizable per character.
  - some builds may share the same gear. e.g. you may have 2 power builds: both use the same full berserker gear template tab but different build template tabs. Or the other way around: same build template tab but different gear templates. 
  - a switch to a gear template tab and build template tab at the same time can be achieved by triggering the gw2 hotkeys for the gear tab and the build tab. This may require checking whether the focus is currently on the gw2 ingame chat or in some other text input field in gw2 or blish hud. because then the hotkeys may not switch the build/gear but instead accidently type into those text inputs. That should be prevented e.g. by not triggering the gear/build tab switch or by other means.
  - niche case:  a user may want to switch only the gear tab OR the equip tab, not both.
  - wvw and pve share the same gear/build templates. But spvp has other gear templates than wvw/pve
  
  
was requested in discord by a user: https://discord.com/channels/531175899588984842/534490472224391169/1098093153728348301

**Comment by KenediaDev** [comment]
I've been thinking about adding something like that to builds manager :) it's a loose idea but I think it would fit very well
